geomstats/geometry/pullback_metric.py

```diff
@@ -103,6 +103,415 @@ def pickable_inner_product(i, j):
+            Point.
+
+        Returns
+        -------
+        log : array-like
+            Logarithm of point from base_point.
+        """
+        new_base_point = self.diffeomorphism(base_point)
+        new_point = self.diffeomorphism(point)
+        new_log = self.embedding_metric.log(new_point, new_base_point, **kwargs)
+        log = self.inverse_tangent_diffeomorphism(new_log, new_base_point)
+        return log
+
+    def dist(self, point_a, point_b, **kwargs):
+        """
+        Compute the distance via diffeomorphic pullback.
+
+        Parameters
+        ----------
+        point_a : array-like
+            Point a.
+        point_b : array-like
+            Point b.
+
+        Returns
+        -------
+        dist : array-like
+            Distance between point_a and point_b.
+        """
+        new_point_a = self.diffeomorphism(point_a)
+        new_point_b = self.diffeomorphism(point_b)
+        new_distance = self.embedding_metric.dist(new_point_a, new_point_b, **kwargs)
+        return new_distance
+
+    def curvature(self, tangent_vec_a, tangent_vec_b, tangent_vec_c, base_point):
+        """
+        Compute the curvature via diffeomorphic pullback.
+
+        Parameters
+        ----------
+        tangent_vec_a : array-like
+            Tangent vector a.
+        tangent_vec_b : array-like
+            Tangent vector b.
+        tangent_vec_c : array-like
+            Tangent vector c.
+        base_point : array-like
+            Base point.
+
+        Returns
+        -------
+        curvature : array-like
+            Curvature in directions tangent_vec a, b, c at base_point.
+        """
+        new_base_point = self.diffeomorphism(base_point)
+        new_tangent_vec_a = self.tangent_diffeomorphism(tangent_vec_a, base_point)
+        new_tangent_vec_b = self.tangent_diffeomorphism(tangent_vec_b, base_point)
+        new_tangent_vec_c = self.tangent_diffeomorphism(tangent_vec_c, base_point)
+        new_curvature = self.embedding_metric.curvature(
+            new_tangent_vec_a, new_tangent_vec_b, new_tangent_vec_c, new_base_point
+        )
+        curvature = self.inverse_tangent_diffeomorphism(new_curvature, new_base_point)
+        return curvature
+
+    def parallel_transport(
+        self, tangent_vec, base_point, direction=None, end_point=None
+    ):
+        """
+        Compute the parallel transport via diffeomorphic pullback.
+
+        Parameters
+        ----------
+        tangent_vec : array-like
+            Tangent vector.
+        base_point : array-like
+            Base point.
+        direction : array-like
+            Direction.
+        end_point : array-like
+            End point.
+
+        Returns
+        -------
+        parallel_transport : array-like
+            Parallel transport.
+        """
+        new_base_point = self.diffeomorphism(base_point)
+        new_tangent_vec = self.tangent_diffeomorphism(tangent_vec, base_point)
+        if direction is None:
```

Comment:
Maybe we can use the same name here to simplify branching:

```
if direction is not None:
  direction = self.tangent_diffeomorphism(direction, base_point)
```

Reply:
Actually I decided to live it to have consistant naming for quantity for the metric or the image metric